### ISSUE TITLE
Previous stage(s) lost when I stage lines with integrated Git.

### ISSUE THREAD

**Issue by 3ToThe6** [Microsoft/vscode#66027]
Issue Type: <b>Bug</b>

Here's how this issue can be reproduced:

1. `mkdir a`

2. `cd a`

3. `git init`

4. `mkdir b`

5. Create a file `b.txt` in dir `b`. It has three lines, `a`, `b` and an empty one, with CRLF `'\n'`.

6. `git add b/b.txt`

7. `git commit -m I`

8. `code b --disable-extensions` (Not `code .`)

9. Open `b.txt` with Code. First change the first line to `c`, and then stage it. Second change the second line to `d`, and stage it.

It seems that the first stage was discarded:
<img width="771" alt="code" src="https://user-images.githubusercontent.com/32224706/50686457-84a3bb00-1058-11e9-9716-19d188699287.PNG">

VS Code version: Code 1.30.1 (dea8705087adb1b5e5ae1d9123278e178656186a, 2018-12-18T18:12:07.165Z)
OS version: Windows_NT x64 10.0.17134

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-7500U CPU @ 2.70GHz (4 x 2904)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: enabled<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Memory (System)|7.85GB (2.48GB free)|
|Process Argv|. --disable-extensions|
|Screen Reader|no|
|VM|0%|

</details>Extensions disabled

**Comment by joaomoreno** [comment]
After step 9, since you staged all changes, there should only be a (1) in the SCM count badge, although I see a (2) in your screenshot. What does the SCM viewlet display?

**Comment by 3ToThe6** [comment]
<img width="774" alt="a" src="https://user-images.githubusercontent.com/32224706/50764463-9d58de80-12ad-11e9-9add-646abb659855.PNG">
<img width="774" alt="b" src="https://user-images.githubusercontent.com/32224706/50764465-a2b62900-12ad-11e9-8f6a-4f4c8ff85329.PNG">
I did `git diff` and `git diff --cached` in terminal, and got:
<img width="914" alt="c" src="https://user-images.githubusercontent.com/32224706/50764496-c37e7e80-12ad-11e9-8ec6-54de147a0267.PNG">
I restarted Code and could get:
<img width="774" alt="d" src="https://user-images.githubusercontent.com/32224706/50764551-fc1e5800-12ad-11e9-9438-2345391d7e2d.PNG">
It's different from the first screenshot.

**Comment by joaomoreno** [comment]
Can you reproduce if you open folder `a` instead of folder `b` in Code?

**Comment by 3ToThe6** [comment]
No. The issue only occurs when I open folder `b`. I also did a new test in which file name `b.txt` was replaced by `c.txt`, and this issue could also be reproduced.

**Comment by joaomoreno** [comment]
Duplicate of #41085, blocked by #3025
